File: Backend/TaskCollaborationApp.API/Services/UserService.cs
using TaskCollaborationApp.API.Controllers.DTOs.User;
using TaskCollaborationApp.API.Repositories.Interfaces;
using TaskCollaborationApp.API.Services.Interfaces;

namespace TaskCollaborationApp.API.Services
{
    /// <summary>
    /// User service implementation.
    /// Handles user retrieval operations for task assignment.
    /// </summary>
    public class UserService : IUserService
    {
        private readonly IUnitOfWork _unitOfWork;

        public UserService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }

        /// <inheritdoc />
        public async Task<IEnumerable<UserListItemDto>> GetAllUsersAsync()
        {
            var users = await _unitOfWork.Users.GetAllAsync();

            return users.Select(u => new UserListItemDto
            {
                Id = u.Id,
                Name = u.Name,
                Initials = GetInitials(u.Name)
            });
        }

        /// <inheritdoc />
        public async Task<UserResponseDto?> GetUserByIdAsync(int id)
        {
            var user = await _unitOfWork.Users.GetByIdAsync(id);

            if (user == null)
                return null;

            return new UserResponseDto
            {
                Id = user.Id,
                Name = user.Name,
                Email = user.Email,
                Initials = GetInitials(user.Name),
                CreatedAt = user.CreatedAt
            };
        }

        /// <summary>
        /// Generates initials from a full name.
        /// Example: "John Doe" -> "JD"
        /// </summary>
        private static string GetInitials(string name)
        {
            var parts = name.Split(' ', StringSplitOptions.RemoveEmptyEntries);
            if (parts.Length == 0) return "?";
            if (parts.Length == 1) return parts[0][0].ToString().ToUpper();
            return $"{parts[0][0]}{parts[^1][0]}".ToUpper();
        }
    }
}

File: Frontend/TaskCollaborationApp.Client/src/features/user/services/userService.ts
import api from "../../../services/api";
import type { UserListItemDto } from "../types/api.types";

/**
 * userService - User 관련 API 호출을 담당
 *
 * Client 활용:
 * - TaskForm에서 Assignee 드롭다운 데이터 로드
 * - 실제 HTTP 요청을 보내고 응답을 반환
 */
export const userService = {
  /**
   * getAllUsers - 전체 사용자 목록 조회
   *
   * Client 활용:
   * - TaskForm에서 컴포넌트 마운트 시 호출
   * - Assignee 드롭다운 옵션으로 사용
   *
   * @returns 사용자 목록 (id, name, initials)
   */
  getAllUsers: async (): Promise<UserListItemDto[]> => {
    const response = await api.get<UserListItemDto[]>("/users");
    return response.data;
  },
};
